app/controllers/glossary_terms_controller.rb

```diff
@@ -80,6 +75,47 @@ def destroy
 
   private
 
+  # --------- index private methods
+
+  # should index be filtered?
+  def filter_index?
+    params[:q] || params[:by]
+  end
+
+  def index_filtered
+    query = find_or_create_query(:GlossaryTerm, by: params[:by])
+    show_selected_articles(
+      query,
+      id: params[:id].to_s,
+      always_index: true
+    )
+  end
+
```

Comment:
@JoeCohen - Is this necessary? I didn't include any filtering tabs on the index and now that I think about it, it seems not. I also can't imagine that a `:q` param would be relevant to a glossary term index. We basically just want pagination and pattern search, I reckon.

Reply:
Maybe this also needs to handle pattern searches explicitly. Looking to `articles_controller` for guidance... I don't find an index action for the `pattern` param in that controller, although there is an `Query::ArticlePatternSearch` class. 

`Articles` are also not available in the search bar dropdown menu. This may be an unimplemented feature for articles.

Reply:
What you're saying about filtering, :by, and :q makes sense. 
Perhaps in the future we could have filtering, by, and q. But in this PR  I think they should be omitted.

It's been so long since I wrote Articles that I don't remember anything about why I wrote Articles the way I did.
It was probably a combo of copying some example from the cloud and groping in the dark.

Reply:
Ok, thanks. In a separate PR I may remove the pattern search query class from Articles since it is as yet unimplemented, what do you think? Or,  I could implement it. But it doesnt seem that useful. 

I'm glad it was there for me to copy here, though!